src/ts/content/extensions/enforceAuthAndTagIndex.ts

```diff
@@ -0,0 +1,37 @@
+import {ConfigKey} from "../../common/entities/config";
+import config from "../../common/entities/config";
+import {onBackgroundEvent, onceBackgroundEvent} from "../util/onBackgroundEvent";
+import {BackgroundEvent} from "../../common/backgroundEvent";
+import {isAuthorized} from "./author/util/isAuthorized";
+import {createModal} from "../../common/ui/modal";
+import {UIColour} from "../../common/ui/colour";
+
+const isSpreadSheetLinkSet = async () => config.get(ConfigKey.SpreadsheetLink);
+
+window.addEventListener("pageshow", async () => {
+	if (await config.get(ConfigKey.EnforceAuthAndTagIndex)) {
+		if (!(await isAuthorized()) || !(await isSpreadSheetLinkSet())) {
+			const exit = createModal({
+				text: "Hey there!",
+				subText: [
+					"It looks like you might not be logged in or do not have a tag index set",
+					"Please login and/or set a tag index link under the extension options",
+				],
+				colour: UIColour.PURPLE,
+				exitable: false,
+			});
+
+			onBackgroundEvent(BackgroundEvent.CompletedAuth, async () => {
+				if (await isSpreadSheetLinkSet()) {
+					exit();
+				}
+			});
+
+			onceBackgroundEvent(BackgroundEvent.AddedSheetLink, async () => {
```

Comment:
still not super sure why we want `onceBackgroundEvent` for the tag index here and `onBackgroundEvent` for auth

Reply:
yeah i don't think this implementation needs `once` specifically... I had something specific in my brain at the time I was talking about it involving loaders etc that would have been TERRIBLE for the user